table/block.cc

```diff
@@ -283,9 +294,33 @@ bool BlockIter::ParseNextKey() {
 
     value_ = Slice(p + non_shared, value_length);
     while (restart_index_ + 1 < num_restarts_ &&
-           GetRestartPoint(restart_index_ + 1) < current_) {
+           GetRestartPoint(restart_index_ + 1) <= current_) {
       ++restart_index_;
     }
+    if (value_delta_encoded_) {
+      assert(value_length == 0);
+      if (GetRestartPoint(restart_index_) == current_) {
+        const size_t kHandleMaxSize = 16;
+        auto v = Slice(p + non_shared, kHandleMaxSize);
+        BlockHandle handle;
+        Status decode_s = handle.DecodeFrom(&v);
+        decoded_value_ = handle;
+        auto& updated_value = v;
+        assert(decode_s.ok());
+        value_ = Slice(p + non_shared, updated_value.data() - value_.data());
+      } else {
+        const size_t kHandleMaxSize = 8;
+        auto v = Slice(p + non_shared, kHandleMaxSize);
+        auto next_value_base =
+            decoded_value_.offset() + decoded_value_.size() + kBlockTrailerSize;
+        BlockHandle handle;
+        Status decode_s = handle.DecodeSizeFrom(next_value_base, &v);
+        decoded_value_ = handle;
+        auto& updated_value = v;
```

Comment:
This `auto` also makes the code hard for me to understand. I suggest we make it explicit.

Reply:
Sure..